domain/blockdag/process.go

```diff
@@ -226,124 +228,225 @@ func (dag *BlockDAG) createBlockNodeFromBlock(block *util.Block) (
 // connectBlock handles connecting the passed node/block to the DAG.
 //
 // This function MUST be called with the DAG state lock held (for writes).
-func (dag *BlockDAG) connectBlock(node *blockNode,
-	block *util.Block, selectedParentAnticone []*blockNode, flags BehaviorFlags) (*chainUpdates, error) {
+func (dag *BlockDAG) connectBlock(newNode *blockNode,
+	block *util.Block, selectedParentAnticone []*blockNode, flags BehaviorFlags) (*selectedParentChainUpdates, error) {
 
-	err := dag.checkBlockTransactionsFinalized(block, node, flags)
+	err := newNode.checkDAGRelations()
 	if err != nil {
 		return nil, err
 	}
 
-	if err = dag.checkFinalityViolation(node); err != nil {
+	err = dag.checkBlockTransactionsFinalized(block, newNode, flags)
+	if err != nil {
+		return nil, err
+	}
+
+	err = dag.checkBlockHasNoChainedTransactions(block, newNode, flags)
+	if err != nil {
 		return nil, err
 	}
 
 	if err := dag.validateGasLimit(block); err != nil {
 		return nil, err
 	}
 
-	newBlockPastUTXO, txsAcceptanceData, newBlockFeeData, newBlockMultiSet, err :=
-		node.verifyAndBuildUTXO(dag, block.Transactions(), isBehaviorFlagRaised(flags, BFFastAdd))
+	var isNewSelectedTip bool
+	if newNode.isGenesis() {
+		isNewSelectedTip = true
+	} else {
+		// If the new block is not the selected tip - it's not in virtual's selectedParentChain, therefore it's
+		// not going to be utxo-verified at this point
+		isNewSelectedTip = dag.selectedTip().less(newNode)
+		if !isNewSelectedTip {
+			dag.index.SetBlockNodeStatus(newNode, statusUTXONotVerified)
+		}
+	}
+
+	dbTx, err := dag.databaseContext.NewTx()
+	if err != nil {
+		return nil, err
+	}
+	defer dbTx.RollbackUnlessClosed()
+
+	if isNewSelectedTip {
+		err = dag.validateAndApplyUTXOSet(newNode, block, dbTx)
+		if err != nil {
+			if !errors.As(err, &(RuleError{})) {
+				return nil, err
+			}
+			dag.index.SetBlockNodeStatus(newNode, statusDisqualifiedFromChain)
```

Comment:
I don't think this is newsworthy enough of a Warn log.

I mean, we don't care about block UTXO-validation enough to validate all blocks, but then if a block had the luck to get validated and wasn't found valid, we WARN about this? Seems a bit wrong for me.

Nevertheless, I agree that for debugging this would be useful.
Therefore, I've added to `log.Debugf("Accepted block blabla")` the status of the new block.
Note that to achieve that I had to move the log into maybeAcceptBlock, which is more then OK in my opinion.